packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -40,26 +40,21 @@ export class H5WasmApi extends ProviderApi {
   }
 
   public async getEntity(path: string): Promise<Entity> {
-    const file = await this.file;
-    const h5wEntity = file.get(path);
-
+    const h5wEntity = await this.getH5WasmEntity(path);
     return this.processH5WasmEntity(getNameFromPath(path), path, h5wEntity);
   }
 
   public async getValue(params: ValuesStoreParams): Promise<unknown> {
     const { dataset } = params;
 
-    const file = await this.file;
-    const h5wDataset = file.get(dataset.path);
+    const h5wDataset = await this.getH5WasmEntity(dataset.path);
     assertH5WasmDataset(h5wDataset);
 
-    const array = h5wDataset.value;
-    return hasScalarShape(dataset) ? array[0] : array;
+    return h5wDataset.value;
```

Comment:
The new, stricter type of `value` allowed me to identify a bug here: this logic is not needed since h5wasm returns scalars directly.

Reply:
This fixes the issue with H5Web showing only the first character for scalar datasets, like `{` for datasets containing JSON.